packages/vite/src/node/utils.ts

```diff
@@ -815,6 +815,43 @@ export async function getLocalhostAddressIfDiffersFromDNS(): Promise<
   return isSame ? undefined : nodeResult.address
 }
 
+export function diffDnsOrderChange(
+  oldUrls: ViteDevServer['resolvedUrls'],
+  newUrls: ViteDevServer['resolvedUrls'],
+): boolean {
+  if (oldUrls && newUrls) {
+    const { local: oldLocal, network: oldNetwork } = oldUrls
+    const { local: newLocal, network: newNetwork } = newUrls
+    const localChange = oldLocal.length !== newLocal.length
+    const networkChange = oldNetwork.length !== newNetwork.length
+    if (!localChange) {
+      for (let i = 0; i < oldLocal.length; i++) {
+        if (oldLocal[i] !== newLocal[i]) {
+          return true
+        }
+      }
+    }
+    if (!networkChange) {
+      for (let i = 0; i < oldNetwork.length; i++) {
+        if (oldNetwork[i] !== newNetwork[i]) {
+          return true
+        }
+      }
+    }
+    return localChange || networkChange
+  }
+  if (!oldUrls && !newUrls) {
+    return false
+  }
+  if (oldUrls && !newUrls) {
+    return true
+  }
+  if (!oldUrls && newUrls) {
+    return true
+  }
+  return false
```

Comment:
You can use `arrayEqual` from utils to simplify this:
```js
return !(oldUrls === newUrls || (arrayEqual(oldUrls.local, newUrls.local) && arrayEqual(oldUrls.network, newUrls.network)))
```

Reply:
It looks great 👍